src/test/tests/scanAnlayzerRunner.test.ts

```diff
@@ -0,0 +1,240 @@
+        return {
+            get url() {
+                return inputUrl;
+            },
+            get username() {
+                return user;
+            },
+            get password() {
+                return pass;
+            },
+            get accessToken() {
+                return token;
+            },
+            areXrayCredentialsSet(): boolean {
+                return !!(inputUrl && ((user && pass) || token));
+            }
+        } as ConnectionManager;
+    }
+
+    function createDummyBinaryRunner(
+        connection: ConnectionManager = connectionManager,
+        timeout: number = ScanUtils.ANALYZER_TIMEOUT_MILLISECS,
+        dummyAction: () => Promise<void> = () => Promise.resolve()
+    ): BinaryRunner {
+        return new (class extends BinaryRunner {
+            // eslint-disable-next-line @typescript-eslint/no-unused-vars
+            async runBinary(
+                checkCancel: () => void,
+                // eslint-disable-next-line @typescript-eslint/no-unused-vars
+                _yamlConfigPath: string,
+                // eslint-disable-next-line @typescript-eslint/no-unused-vars
+                _executionLogDirectory: string
+            ): Promise<void> {
+                await RunUtils.runWithTimeout(timeout, checkCancel, dummyAction());
+            }
+        })(connection, timeout, dummyName, logManager);
+    }
+
+    [
+        {
+            name: 'Credentials are not set',
+            shouldFail: true,
+            url: '',
+            user: '',
+            pass: '',
+            token: '',
+            proxy: undefined,
+            logPath: undefined
+        },
+        {
+            name: 'Password credentials',
+            shouldFail: false,
+            url: 'platformUrl',
+            user: 'username',
+            pass: 'password',
+            token: '',
+            proxy: undefined,
+            logPath: undefined
+        },
+        {
+            name: 'Access token credentials',
+            shouldFail: false,
+            url: 'platformUrl',
+            user: '',
+            pass: '',
+            token: 'access-token',
+            proxy: undefined,
+            logPath: undefined
+        },
+        {
+            name: 'With log path',
+            shouldFail: false,
+            url: 'platformUrl',
+            user: '',
+            pass: '',
+            token: 'access-token',
+            proxy: undefined,
+            logPath: 'path/to/folder'
+        },
+        {
+            name: 'With proxy env vars',
+            shouldFail: false,
+            url: 'platformUrl',
+            user: '',
+            pass: '',
+            token: 'access-token',
+            proxy: 'proxyUrlEnvVarValue',
+            logPath: undefined
+        }
+    ].forEach(test => {
+        it('Create environment variables for binary execution - ' + test.name, () => {
```

Comment:
`it` title should be clear about what is being tested and/or the expected results